stan/math/rev/functor/reduce_sum.hpp

```diff
@@ -216,12 +269,14 @@ struct reduce_sum_impl<ReduceFunction, require_var_t<ReturnType>, ReturnType,
     save_varis(varis, vmapped);
     save_varis(varis + num_vars_sliced_terms, args...);
 
-    for (size_t i = 0; i < num_vars_sliced_terms; ++i) {
-      partials[i] = 0.0;
+    Eigen::VectorXd shared_terms_adjoints(num_vars_shared_terms);
```

Comment:
I like backup or something in the name here too -- like `backup_shared_adjoints`.

Reply:
Sure.